api/internal/localizer/localizer_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer //nolint:testpackage
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/kustomize/api/hasher"
+	"sigs.k8s.io/kustomize/api/internal/plugins/loader"
+	"sigs.k8s.io/kustomize/api/internal/target"
+	"sigs.k8s.io/kustomize/api/internal/validate"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func makeMemoryFs(t *testing.T) filesys.FileSystem {
+	t.Helper()
+	req := require.New(t)
+
+	fSys := filesys.MakeFsInMemory()
+	req.NoError(fSys.MkdirAll("/a/b"))
+	req.NoError(fSys.WriteFile("/a/pod.yaml", []byte("pod configuration")))
```

Comment:
I think these tests would be easier to read if we have an actual pod configuration here. As I was looking through the tests, my first reaction was wondering what does this string "pod configuration" mean and where did it come from. We are also using this string 7 times, so we should probably have a const for it.

You can have some const at the top

```
const podConfiguration = `
apiVersion: v1
kind: Pod
metadata:
  name: nginx
`
```

and then use the `podConfiguration` const throughout these tests.